### DISCUSSION TITLE
Application Factory guidance

### DISCUSSION THREAD

**Discussion by fluffy-cakes** [tiangolo/uwsgi-nginx-flask-docker#372] [Questions] [answered]
I have created an app very similar to this tutorial which uses Application Factory: https://github.com/pallets/flask/tree/2.0.2/examples/tutorial . Which is taken from Flask website itself: https://flask.palletsprojects.com/en/2.0.x/tutorial/ . My app doesn't necessarily use nested `flaskr` folder like in this example, but I guess it can be. 

I've had a good web search on how to apply your container using Application Factory, but I can't seem to find anything that relates to using it with the `uwsgi.ini` file. I have seen this closed task, but without his original code it's hard to understand how it was/should be laid out: https://github.com/tiangolo/uwsgi-nginx-flask-docker/issues/105

Using the tutorial example from the first tutorial link above, can you please explain how to use this container along with Application Factory?

FYI, I too get the following error `unable to load app 0 (mountpoint='') (callable not found or import error)`, and I also use `create_app()` function to nest the app itself: `app = flask.Flask(__name__)`. Basically, almost identical to that original tutorial.

**Comment by tiangolo** [comment, marked as answer]
It's a bit difficult without looking at your code, but you could create a file with something like `app/main.py` that does something like this:

```Python
from .some_module import create_app

app = create_app()
```
